src/test/app/Credentials_test.cpp

```diff
@@ -0,0 +1,949 @@
+                BEAST_EXPECT(
+                    jle.isObject() && jle.isMember(jss::result) &&
+                    jle[jss::result].isMember(jss::error) &&
+                    jle[jss::result][jss::error] == "entryNotFound");
+            }
+        }
+
+        {
+            testcase("Create for themself.");
+
+            auto const credKey = credentialKeylet(issuer, issuer, credType);
+
+            env(credentials::create(issuer, issuer, credType),
+                credentials::uri(uri));
+            env.close();
+            {
+                auto const sleCred = env.le(credKey);
+                BEAST_EXPECT(static_cast<bool>(sleCred));
+                if (!sleCred)
+                    return;
+
+                BEAST_EXPECT(sleCred->getAccountID(sfSubject) == issuer.id());
+                BEAST_EXPECT(sleCred->getAccountID(sfIssuer) == issuer.id());
+                BEAST_EXPECT((sleCred->getFieldU32(sfFlags) & lsfAccepted));
+                BEAST_EXPECT(
+                    sleCred->getFieldU64(sfIssuerNode) ==
+                    sleCred->getFieldU64(sfSubjectNode));
+                BEAST_EXPECT(ownerCount(env, issuer) == 1);
+                BEAST_EXPECT(checkVL(sleCred, sfCredentialType, credType));
+                BEAST_EXPECT(checkVL(sleCred, sfURI, uri));
+                auto const jle =
+                    credentials::ledgerEntry(env, issuer, issuer, credType);
+                BEAST_EXPECT(
+                    jle.isObject() && jle.isMember(jss::result) &&
+                    !jle[jss::result].isMember(jss::error) &&
+                    jle[jss::result].isMember(jss::node) &&
+                    jle[jss::result][jss::node].isMember("LedgerEntryType") &&
+                    jle[jss::result][jss::node]["LedgerEntryType"] ==
+                        jss::Credential);
+            }
+
+            env(credentials::deleteCred(issuer, issuer, issuer, credType));
+            env.close();
+            {
+                BEAST_EXPECT(!env.le(credKey));
+                BEAST_EXPECT(!ownerCount(env, issuer));
+
+                // check no credential exists anymore
+                auto const jle =
+                    credentials::ledgerEntry(env, issuer, issuer, credType);
+                BEAST_EXPECT(
+                    jle.isObject() && jle.isMember(jss::result) &&
+                    jle[jss::result].isMember(jss::error) &&
+                    jle[jss::result][jss::error] == "entryNotFound");
+            }
+        }
+    }
+
+    void
+    testCredentialsDelete(FeatureBitset features)
+    {
+        using namespace test::jtx;
+
+        const char credType[] = "abcde";
+
+        Account const issuer{"issuer"};
+        Account const subject{"subject"};
+        Account const other{"other"};
+
+        Env env{*this, features};
+
+        // fund subject and issuer
+        env.fund(XRP(5000), issuer, subject, other);
+        env.close();
+
+        {
```

Comment:
Aren't all these delete tests already in `AccountDelete_test`?

Reply:
No, cause these tests  mainly check deletion of the credential object, not the account itself